src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -0,0 +1,858 @@
+                                                                                         equations)
+
+    mag_en = 0.5f0 * (B1^2 + B2^2 + B3^2)
+    div_clean_energy = 0.5f0 * psi^2
+
+    f = zero(MVector{nvariables(equations), eltype(u)})
+
+    if orientation == 1
+        f[1] = equations.c_h * psi
+        f[2] = v1_plus * B2 - v2_plus * B1
+        f[3] = v1_plus * B3 - v3_plus * B1
+
+        for k in eachcomponent(equations)
+            rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, u, equations)
+            v1 = rho_v1 / rho
+            v2 = rho_v2 / rho
+            v3 = rho_v3 / rho
+            kin_en = 0.5f0 * rho * (v1^2 + v2^2 + v3^2)
+
+            gamma = equations.gammas[k]
+            p = (gamma - 1) * (rho_e - kin_en - mag_en - div_clean_energy)
+
+            f1 = rho_v1
+            f2 = rho_v1 * v1 + p
+            f3 = rho_v1 * v2
+            f4 = rho_v1 * v3
+            f5 = (kin_en + gamma * p / (gamma - 1)) * v1 + 2 * mag_en * vk1_plus[k] -
+                 B1 * (vk1_plus[k] * B1 + vk2_plus[k] * B2 + vk3_plus[k] * B3) +
+                 equations.c_h * psi * B1
+
+            set_component!(f, k, f1, f2, f3, f4, f5, equations)
+        end
+        f[end] = equations.c_h * B1
+    else #if orientation == 2
+        f[1] = v2_plus * B1 - v1_plus * B2
+        f[2] = equations.c_h * psi
+        f[3] = v2_plus * B3 - v3_plus * B2
+
+        for k in eachcomponent(equations)
+            rho, rho_v1, rho_v2, rho_v3, rho_e = get_component(k, u, equations)
+            v1 = rho_v1 / rho
+            v2 = rho_v2 / rho
+            v3 = rho_v3 / rho
+            kin_en = 0.5f0 * rho * (v1^2 + v2^2 + v3^2)
+
+            gamma = equations.gammas[k]
+            p = (gamma - 1) * (rho_e - kin_en - mag_en - div_clean_energy)
+
+            f1 = rho_v2
+            f2 = rho_v2 * v1
+            f3 = rho_v2 * v2 + p
+            f4 = rho_v2 * v3
+            f5 = (kin_en + gamma * p / (gamma - 1)) * v2 + 2 * mag_en * vk2_plus[k] -
+                 B2 * (vk1_plus[k] * B1 + vk2_plus[k] * B2 + vk3_plus[k] * B3) +
+                 equations.c_h * psi * B2
+
+            set_component!(f, k, f1, f2, f3, f4, f5, equations)
+        end
+        f[end] = equations.c_h * B2
+    end
+
+    return SVector(f)
+end
+
+"""
+    flux_nonconservative_ruedaramirez_etal(u_ll, u_rr,
+                                               orientation::Integer,
+                                               equations::IdealGlmMhdMultiIonEquations2D)
+
+Entropy-conserving non-conservative two-point "flux"" as described in 
+- A. Rueda-Ramírez, A. Sikstel, G. Gassner, An Entropy-Stable Discontinuous Galerkin Discretization
+  of the Ideal Multi-Ion Magnetohydrodynamics System (2024). Journal of Computational Physics.
+  [DOI: 10.1016/j.jcp.2024.113655](https://doi.org/10.1016/j.jcp.2024.113655).
+
+!!! info "Usage and Scaling of Non-Conservative Fluxes in Trixi"
```

Comment:
```suggestion
!!! info "Usage and Scaling of Non-Conservative Fluxes in Trixi.jl"
```
We try to always use Trixi.jl as the code name, to distinguish it better from other things named "trixi" when people are searching for it 😉